src/utilities/resolveSchemaCoordinate.ts

```diff
@@ -98,84 +98,24 @@ export function resolveSchemaCoordinate(
   );
 }
 
-/**
- * SchemaCoordinate : @ Name
- */
-function resolveDirectiveCoordinate(
-  schema: GraphQLSchema,
-  schemaCoordinate: DirectiveCoordinateNode,
-): ResolvedDirective | undefined {
-  // Let {directiveName} be the value of the first {Name}.
-  const directiveName = schemaCoordinate.name.value;
-
-  // Let {directive} be the directive in the {schema} named {directiveName}.
-  const directive = schema.getDirective(directiveName);
-
-  // If {directive} does not exist, return undefined.
-  if (!directive) {
-    return;
-  }
-
-  // Otherwise return the directive in the {schema} named {directiveName}.
-  return { kind: 'Directive', directive };
-}
-
-/**
- * SchemaCoordinate : @ Name ( Name : )
- */
-function resolveDirectiveArgumentCoordinate(
-  schema: GraphQLSchema,
-  schemaCoordinate: DirectiveArgumentCoordinateNode,
-): ResolvedDirectiveArgument | undefined {
-  // Let {directiveName} be the value of the first {Name}.
-  const directiveName = schemaCoordinate.name.value;
-
-  // Let {directive} be the directive in the {schema} named {directiveName}.
-  const directive = schema.getDirective(directiveName);
-
-  // Assert that {directive} exists.
-  if (!directive) {
-    throw new Error(
-      `Expected ${inspect(directiveName)} to be defined as a directive in the schema.`,
-    );
-  }
-
-  // Let {directiveArgumentName} be the value of the second {Name}.
-  const {
-    argumentName: { value: directiveArgumentName },
-  } = schemaCoordinate;
-  const directiveArgument = directive.args.find(
-    (arg) => arg.name === directiveArgumentName,
-  );
-
-  // If {directiveArgumentName} does not exist, return undefined.
-  if (!directiveArgument) {
-    return;
-  }
-
-  // Return the argument of {directive} named {directiveArgumentName}.
-  return { kind: 'DirectiveArgument', directive, directiveArgument };
-}
-
 /**
  * SchemaCoordinate : Name
  */
 function resolveTypeCoordinate(
   schema: GraphQLSchema,
   schemaCoordinate: TypeCoordinateNode,
 ): ResolvedNamedType | undefined {
-  // Let {typeName} be the value of the first {Name}.
+  // 1. Let {typeName} be the value of the first {Name}.
+  // 2. Let {type} be the type in the {schema} named {typeName}.
   const typeName = schemaCoordinate.name.value;
-
-  // Let {type} be the type in the {schema} named {typeName}.
   const type = schema.getType(typeName);
 
-  // If {type} does not exist, return undefined.
+  // 3. If {type} does not exist, return {void}.
```

Comment:
If this is actual speed text it should say null not void.

Reply:
Ack, will do. Was wondering what to pick here, but I now see that we do that in other places, e.g. 6.4.3, thanks!

(Just curious tho, does this not create there a risk of ambiguity in the reference JS implementation where undefined != null?)

Reply:
(No; such things are for the reference implementation to worry about (via `!= null` rather than `!== null`), the GraphQL spec doesn’t need to cater to JS’s idiosyncrasies.)